<?xml version="1.0"?>
<robot name="materials">

  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>

  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>

  <material name="red">
    <color rgba="0.8 0 0 1"/>
  </material>

  <material name="green">
    <color rgba="0 0.8 0 0.5"/>
  </material>

  <link name="base_link">
  </link>

  <link name="base_link_back">
    <collision>
    <origin rpy="1.57 0 1.57" xyz = "0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://quadruped_simulation/urdf/STL_simulation/back_BODY.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="0.074   0.0   -0.01 "/>
      <mass value="0.881" />
      <inertia ixx="0.002" ixy="4.805E-09" ixz="1.767E-15"
                              iyy="0.003" iyz="-5.737E-10"
                                              izz="0.003" />
    </inertial>
    <visual>
    <origin rpy="1.57 0 1.57" xyz="0 0.0 0"/>
      <geometry>
        <mesh filename="package://quadruped_simulation/urdf/STL_simulation/back_BODY.stl"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>

<joint name="base_to_back" type="fixed">
    <parent link="base_link"/>
    <child link="base_link_back"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
  </joint>

  <link name="base_link_front">
    <collision>
    <origin rpy="1.57 0 1.57" xyz = "0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://quadruped_simulation/urdf/STL_simulation/front_BODY.stl"/>
      </geometry>
    </collision>
    <inertial>
      <origin xyz="-0.074  0.0  -0.007"/>
      <mass value="0.888" />
      <inertia ixx="0.002" ixy="5.060E-09" ixz="7.629E-06"
                              iyy="0.003" iyz="6.457E-10"
                                              izz="0.003" />
    </inertial>
    <visual>
    <origin rpy="1.57 0 1.57" xyz = "0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://quadruped_simulation/urdf/STL_simulation/front_BODY.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

<!--BODY JOINTS-->

<joint name="body" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="200" velocity="0.5"/>
    <parent link="base_link_back"/>
    <child link="base_link_front"/>
    <origin xyz=" 0 0 0"/>
  </joint>



<!--_______________LEGS____________-->


#BACK RIGHT
  <link name="coxa_BR">
    <inertial>
      <origin xyz="-0.116 -0.0 -0.0"/>
      <mass value="0.318" />
      <inertia ixx="1E-15" ixy="1.743E-05"  ixz="-2.075E-06"
                                iyy="1E-15" iyz="-1.739E-06"
                                                 izz="1E-15" />
    </inertial>    
    <visual>
      <origin rpy="1.57 0 1.57" xyz='0.0 0.05 0.037'/>
      <geometry>
        <mesh filename="package://quadruped_simulation/urdf/STL_simulation/BR_coxa.stl"/>
      </geometry>
      <material name="red"/>
    </visual>
  </link>

  <link name="femur_BR">
    <inertial>
      <origin xyz="-0.032  -0.03  -0.03"/>
      <mass value="0.033" />
      <inertia ixx="1.466E-05" ixy="1E-15"  ixz="-1.268E-05"
                                iyy="2.644E-05" iyz="1E-15"
                                                 izz="1.466E-05" />
    </inertial>   
    <visual>
      <origin rpy="0 0.7854 0" xyz='-0.032 -0.035 -0.037'/>  
      <geometry>
        <box size="0.014 0.02 0.09"/>
      </geometry>
      <material name="red"/>
    </visual>
  </link>

  <link name="tibia_BR">
    <inertial>
      <origin xyz="0.032  -0.03  -0.03"/>
      <mass value="0.029" />
      <inertia ixx="1.231E-05" ixy="1E-15" ixz="1.022E-05"
                               iyy="2.123E-05"  iyz="1E-15"
                                               izz="1.144E-05" />
    </inertial>   
    <visual>
      <origin rpy="0 -0.7854 0" xyz="0.032 -0.035 -0.037"/>
      <geometry>
        <box size="0.014 0.02 0.09"/>
      </geometry>
      <material name="red"/>
    </visual>
  </link>

  <link name="foot_BR">
    <collision>
      <!--origin rpy="1.57 0 -1.57"  xyz="-0.097 0.0695 -0.1131"/-->
      <geometry>
        <!--mesh filename="package://quadruped_simulation/urdf/STL_simulation/FR_foot.stl"/-->
        <sphere radius="0.02"/>
      </geometry>
    </collision>
    <contact>
		<friction_anchor/>
		<stiffness value="30000.0"/>
		<damping value="1000.0"/>
		<spinning_friction value=".3"/>
		<lateral_friction value="1"/>
    </contact>
    <inertial>
      <mass value="0.0005" />
      <inertia ixx="1.04E-7" ixy="1.0E-9" ixz="1.0E-9"
                                 iyy="1.04E-7"  iyz="1.0E-9"
                                                 izz="1.04E-7" />
    </inertial>   
    <visual>
      <!--origin rpy="1.57 0 -1.57"  xyz="-0.097 0.0695 -0.1131"/-->
      <geometry>
        <!--mesh filename="package://quadruped_simulation/urdf/STL_simulation/FR_foot.stl"/-->
        <sphere radius="0.02"/>
      </geometry>
      <material name="red"/>
    </visual>
  </link>
  
  <!--joints-->
  <joint name="coxaF_BR" type="revolute">
    <axis xyz="1 0 0"/>
    <parent link="base_link_back"/>
    <child link="coxa_BR"/>
    <origin  rpy="0 0 0" xyz="0 -0.05 -0.037"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
  </joint>

  <joint name="femurF_BR" type="revolute">
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
    <parent link="coxa_BR"/>
    <child link="femur_BR"/>
    <origin rpy="0.0 -0.785 0.0" xyz="-0.123 0 0 "/>
  </joint>

  <joint name="tibiaF_BR" type="revolute">
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
    <parent link="femur_BR"/>
    <child link="tibia_BR"/>
    <origin rpy="0 1.57 0" xyz = "-0.064 0.0 -0.064"/>
  </joint>

  <joint name="footF_BR" type="fixed">
    <parent link="tibia_BR"/>
    <child link="foot_BR"/>
    <origin xyz="0.064 -0.035 -0.064"/>
  </joint>




#BACK LEFT


  <link name="coxa_BL">
    <inertial>
      <origin xyz="-0.116  0.0  -0.0"/>
      <mass value="0.318" />
      <inertia ixx="1E-15" ixy="-1.743E-05"  ixz="-2.075E-06"
                                iyy="1E-15" iyz="1.739E-06"
                                                 izz="1E-15" />
    </inertial>    
    <visual>
      <origin rpy="1.57 0 1.57"  xyz="0 -0.05 0.037"/>
      <geometry>
        <mesh filename="package://quadruped_simulation/urdf/STL_simulation/BL_coxa.stl"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>

  <link name="femur_BL">
    <inertial>
      <origin xyz="-0.032  0.03  -0.03"/>
      <mass value="0.033" />
      <inertia ixx="1.466E-05" ixy="1E-15"  ixz="-1.268E-05"
                                iyy="2.644E-05" iyz="1E-15"
                                                 izz="1.466E-05" />
    </inertial>   
    <visual>
      <origin rpy="0 0.7854 0" xyz='-0.032 0.035 -0.037'/>  
      <geometry>
        <box size="0.014 0.02 0.09"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>

  <link name="tibia_BL">
    <inertial>
      <origin xyz="0.032  0.03  -0.03"/>
      <mass value="0.029" />
      <inertia ixx="1.231E-05" ixy="1.00E-15" ixz="1.022E-05"
                               iyy="2.123E-05"  iyz="1.00E-15"
                                               izz="1.144E-05" />
    </inertial>   
    <visual>
      <origin rpy="0 -0.7854 0" xyz="0.032 0.035 -0.037"/>
      <geometry>
        <box size="0.014 0.02 0.09"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>

  <link name="foot_BL">
    <collision>
      <!--origin rpy="1.57 0 1.57"  xyz="-0.097 -0.0695 -0.1131"/-->
      <geometry>
        <!--mesh filename="package://quadruped_simulation/urdf/STL_simulation/FL_foot.stl"/-->
        <sphere radius="0.02"/>
      </geometry>
    </collision>
    <contact>
		<friction_anchor/>
		<stiffness value="30000.0"/>
		<damping value="1000.0"/>
		<spinning_friction value=".3"/>
		<lateral_friction value="1"/>
    </contact>
    <inertial>
      <mass value="0.0005" />
      <inertia ixx="1.04E-7" ixy="1.0E-9" ixz="1.0E-9"
                                 iyy="1.04E-7"  iyz="1.0E-9"
                                                 izz="1.04E-7" />
    </inertial>   
    <visual>
      <!--origin rpy="1.57 0 1.57"  xyz="-0.097 -0.0695 -0.1131"/-->
      <geometry>
        <!--mesh filename="package://quadruped_simulation/urdf/STL_simulation/FL_foot.stl"/-->
        <sphere radius="0.02"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>
  <!--joints-->
  <joint name="coxaF_BL" type="revolute">
    <axis xyz="1 0 0"/>
    <parent link="base_link_back"/>
    <child link="coxa_BL"/>
    <origin rpy="0 0 0"  xyz="0 0.05 -0.037"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
  </joint>

  <joint name="femurF_BL" type="revolute">
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
    <parent link="coxa_BL"/>
    <child link="femur_BL"/>
    <origin rpy="0 -0.785 0"  xyz="-0.123 0.0 0.0"/>
  </joint>

  <joint name="tibiaF_BL" type="revolute">
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
    <parent link="femur_BL"/>
    <child link="tibia_BL"/>
    <origin rpy="0 1.57 0"  xyz="-0.064 0 -0.064"/>
  </joint>

  <joint name="footF_BL" type="fixed">
    <parent link="tibia_BL"/>
    <child link="foot_BL"/>
    <origin rpy="0 0 0"  xyz="0.064 0.035 -0.064"/>
  </joint>






#FRONT RIGHT





  <link name="coxa_FR">
    <inertial>
      <origin xyz="0.116  -0.0  -0.0"/>
      <mass value="0.318" />
      <inertia ixx="1E-15" ixy="-1.743E-05"  ixz="2.075E-06"
                                iyy="1E-15" iyz="-1.739E-06"
                                                 izz="1E-15" />
    </inertial>    
    <visual>
      <origin rpy="1.57 0 1.57"  xyz="0 0.05 0.037"/>
      <geometry>
        <mesh filename="package://quadruped_simulation/urdf/STL_simulation/FR_coxa.stl"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

  <link name="femur_FR">
    <inertial>
      <origin xyz="0.032  -0.03  -0.03"/>
      <mass value="0.033" />
      <inertia ixx="1.466E-05" ixy="1E-15"  ixz="1.268E-05"
                                iyy="2.644E-05" iyz="1E-15"
                                                 izz="1.466E-05" />
    </inertial>   
    <visual>
      <origin rpy="0 -0.7854 0" xyz='0.032 -0.035 -0.037'/>  
      <geometry>
        <box size="0.014 0.02 0.09"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

  <link name="tibia_FR">
    <inertial>
      <origin xyz="-0.032  -0.03  -0.03"/>
      <mass value="0.029" />
      <inertia ixx="1.231E-05" ixy="1.00E-15" ixz="-1.022E-05"
                               iyy="2.123E-05"  iyz="1.00E-15"
                                               izz="1.144E-05" />
    </inertial>   
    <visual>
      <origin rpy="0 0.7854 0" xyz="-0.032 -0.035 -0.037"/>
      <geometry>
        <box size="0.014 0.02 0.09"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>

  <link name="foot_FR">
    <collision>
      <!--origin rpy="1.57 0 -1.57"  xyz="0.097 0.0695 -0.1131"/-->
      <geometry>
        <!--mesh filename="package://quadruped_simulation/urdf/STL_simulation/BR_foot.stl"/-->
        <sphere radius="0.02"/>
      </geometry>
    </collision>
    <contact>
		<friction_anchor/>
		<stiffness value="30000.0"/>
		<damping value="1000.0"/>
		<spinning_friction value=".3"/>
		<lateral_friction value="1"/>
    </contact>	
    <inertial>
      <mass value="0.0005" />
      <inertia ixx="1.04E-7" ixy="1.0E-9" ixz="1.0E-9"
                                 iyy="1.04E-7"  iyz="1.0E-9"
                                                 izz="1.04E-7" />
    </inertial>   
    <visual>
      <!--origin rpy="1.57 0 -1.57"  xyz="0.097 0.0695 -0.1131"/-->
      <geometry>
        <!--mesh filename="package://quadruped_simulation/urdf/STL_simulation/BR_foot.stl"/-->
        <sphere radius="0.02"/>
      </geometry>
      <material name="white"/>
    </visual>
  </link>
<!--joints-->
  <joint name="coxaF_FR" type="revolute">
    <axis xyz="1 0 0"/>
    <parent link="base_link_front"/>
    <child link="coxa_FR"/>
    <origin rpy="0 0 0"  xyz="0 -0.05 -0.037"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
  </joint>

  <joint name="femurF_FR" type="revolute">
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
    <parent link="coxa_FR"/>
    <child link="femur_FR"/>
    <origin rpy="0 0.785 0"  xyz="0.123 0.0 0.0"/>
  </joint>

  <joint name="tibiaF_FR" type="revolute">
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
    <parent link="femur_FR"/>
    <child link="tibia_FR"/>
    <origin rpy="0 -1.57 0"  xyz="0.064 0 -0.064"/>
  </joint>

  <joint name="footF_FR" type="fixed">
    <parent link="tibia_FR"/>
    <child link="foot_FR"/>
    <origin rpy="0 0 0"  xyz="-0.064 -0.035 -0.064"/>
  </joint>


#FRONT LEFT




  <link name="coxa_FL">
    <inertial>
      <origin xyz="0.116  0.0  -0.0"/>
      <mass value="0.318" />
      <inertia ixx="1E-15" ixy="1.743E-05"  ixz="2.075E-06"
                                iyy="1E-15" iyz="1.739E-06"
                                                 izz="1E-15" />
    </inertial>    
    <visual>
      <origin rpy="1.57 0 1.57"  xyz="0 -0.05 0.037"/>
      <geometry>
        <mesh filename="package://quadruped_simulation/urdf/STL_simulation/FL_coxa.stl"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <link name="femur_FL">
    <inertial>
      <origin xyz="0.032  0.03  -0.03"/>
      <mass value="0.033" />
      <inertia ixx="1.466E-05" ixy="1E-15"  ixz="1.268E-05"
                                iyy="2.644E-05" iyz="1E-15"
                                                 izz="1.466E-05" />
    </inertial>   
    <visual>
      <origin rpy="0 -0.7854 0" xyz='0.032 0.035 -0.037'/>  
      <geometry>
        <box size="0.014 0.02 0.09"/>
      </geometry>
      
      <material name="blue"/>
    </visual>
  </link>

  <link name="tibia_FL">
    <inertial>
      <origin xyz="-0.032 0.03  -0.03"/>
      <mass value="0.029" />
      <inertia ixx="1.231E-05" ixy="1.00E-15" ixz="-1.022E-05"
                               iyy="2.123E-05"  iyz="1.00E-15"
                                               izz="1.144E-05" />
    </inertial>   
    <visual>
      <origin rpy="0 0.7854 0" xyz="-0.032 0.035 -0.037"/>
      <geometry>
        <box size="0.014 0.02 0.09"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <link name="foot_FL">
    <collision>
      <!--origin rpy="1.57 0 1.57"  xyz="0.097 -0.0695 -0.1131"/-->
      <geometry>
        <!--mesh filename="package://quadruped_simulation/urdf/STL_simulation/BL_foot.stl"/-->
        <sphere radius="0.02"/>
      </geometry>
    </collision>
    <contact>
		<friction_anchor/>
		<stiffness value="30000.0"/>
		<damping value="1000.0"/>
		<spinning_friction value=".3"/>
		<lateral_friction value="1"/>
    </contact>
    <inertial>
      <mass value="0.0005" />
      <inertia ixx="1.04E-7" ixy="1.0E-9" ixz="1.0E-9"
                                 iyy="1.04E-7"  iyz="1.0E-9"
                                                 izz="7.04E-7" />
    </inertial>   
    <visual>
      <!--origin rpy="1.57 0 1.57"  xyz="0.097 -0.0695 -0.1131"/-->
      <geometry>
        <!--mesh filename="package://quadruped_simulation/urdf/STL_simulation/BL_foot.stl"/-->
        <sphere radius="0.02"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
<!--joints-->
  <joint name="coxaF_FL" type="revolute">
    <axis xyz="1 0 0"/>
    <parent link="base_link_front"/>
    <child link="coxa_FL"/>
    <origin rpy="0 0 0"  xyz="0 0.05 -0.037"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
  </joint>

  <joint name="femurF_FL" type="revolute">
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
    <parent link="coxa_FL"/>
    <child link="femur_FL"/>
    <origin rpy="0 0.785 0"  xyz="0.123 0.0 0.0"/>
  </joint>

  <joint name="tibiaF_FL" type="revolute">
    <axis xyz="0 1 0"/>
    <limit lower="-1.57" upper="1.57" effort="100" velocity="1.0"/>
    <parent link="femur_FL"/>
    <child link="tibia_FL"/>
    <origin rpy="0 -1.57 0"  xyz="0.064 0.0 -0.064"/>
  </joint>

  <joint name="footF_FL" type="fixed">
    <parent link="tibia_FL"/>
    <child link="foot_FL"/>
    <origin rpy="0 0 0"  xyz="-0.064 0.035 -0.064"/>
  </joint>
</robot>








